dora/core/client/fs/src/main/java/alluxio/client/file/DoraFileOutStream.java

```diff
@@ -203,34 +208,60 @@ public void write(byte[] b, int off, int len) throws IOException {
   }
 
   private void writeInternal(int b) throws IOException {
-    if (mWriteToAlluxio) {
-      Integer intVal = b;
-      byte[] bytes = new byte[]{intVal.byteValue()};
-      mNettyDataWriter.writeChunk(bytes, 0, 1);
-      Metrics.BYTES_WRITTEN_ALLUXIO.inc();
-    }
+    try {
+      if (mWriteToAlluxio) {
+        Integer intVal = b;
+        byte[] bytes = new byte[] {intVal.byteValue()};
+        mNettyDataWriter.writeChunk(bytes, 0, 1);
+        Metrics.BYTES_WRITTEN_ALLUXIO.inc();
+      }
 
-    if (mUnderStorageType.isSyncPersist()) {
-      mUnderStorageOutputStream.write(b);
-      Metrics.BYTES_WRITTEN_UFS.inc();
+      if (mUnderStorageType.isSyncPersist()) {
+        mUnderStorageOutputStream.write(b);
+        Metrics.BYTES_WRITTEN_UFS.inc();
+      }
+      mBytesWritten++;
+    } catch (IOException e) {
+      Throwable throwable = mNettyDataWriter.getPacketWriteException();
+      if (throwable != null) {
+        if (e.getMessage() != null) {
+          throw new IOException(throwable.getMessage() + " " + e.getMessage());
+        } else {
+          throw new IOException(throwable.getMessage());
+        }
+      } else {
+        throw e;
+      }
     }
-    mBytesWritten++;
   }
 
   private void writeInternal(byte[] b, int off, int len) throws IOException {
-    Preconditions.checkArgument(b != null, PreconditionMessage.ERR_WRITE_BUFFER_NULL);
-    Preconditions.checkArgument(off >= 0 && len >= 0 && len + off <= b.length,
-        PreconditionMessage.ERR_BUFFER_STATE.toString(), b.length, off, len);
+    try {
+      Preconditions.checkArgument(b != null, PreconditionMessage.ERR_WRITE_BUFFER_NULL);
+      Preconditions.checkArgument(off >= 0 && len >= 0 && len + off <= b.length,
+          PreconditionMessage.ERR_BUFFER_STATE.toString(), b.length, off, len);
 
-    if (mWriteToAlluxio) {
-      mNettyDataWriter.writeChunk(b, off, len);
-      Metrics.BYTES_WRITTEN_ALLUXIO.inc(len);
-    }
-    if (mUnderStorageType.isSyncPersist()) {
-      mUnderStorageOutputStream.write(b, off, len);
-      Metrics.BYTES_WRITTEN_UFS.inc(len);
+      if (mWriteToAlluxio) {
+        mNettyDataWriter.writeChunk(b, off, len);
+        Metrics.BYTES_WRITTEN_ALLUXIO.inc(len);
+      }
+      if (mUnderStorageType.isSyncPersist()) {
+        mUnderStorageOutputStream.write(b, off, len);
+        Metrics.BYTES_WRITTEN_UFS.inc(len);
+      }
+      mBytesWritten += len;
+    } catch (IOException e) {
+      Throwable throwable = mNettyDataWriter.getPacketWriteException();
+      if (throwable != null) {
+        if (e.getMessage() != null) {
```

Comment:
we would lose both e&throwable's stack trace here which might be hard to reason the problem if we encounter any exception

Reply:
You're right. I updated the codes.